src/ng/filter/orderBy.js

```diff
@@ -155,7 +155,7 @@
 orderByFilter.$inject = ['$parse'];
 function orderByFilter($parse) {
   return function(array, sortPredicate, reverseOrder) {
-    if (!(isArrayLike(array))) return array;
+    if (!(isArrayLike(array))) throw minErr('filter')('notarray', 'Expected array but received: {0}', array);
```

Comment:
Sadly this error code is specific to the `filterFilter` so we have to provide a new one and associated documentation in `docs/content/error` or alternatively update the documentation to make this error code more generic.